Updating to add context about how product and engineering work with Use cases (#2422)
sourcegraph/handbook: content/strategy-goals/strategy/working-with-use-cases.md

Sourcegraph has a shared set of [use cases](index.md#use-cases) that flow through much of what we do, and this page is intended to describe how we use them.

# What is a “use case”?

> At Sourcegraph, we aim to understand software development workflows and what matters most so we make decisions about how to best serve our customers. Think of a use case as a common scenario or problem for which Sourcegraph brings forth solutions, they are the value we bring to our customers by solving a particular challenge. It is what they experience or do on our platform that makes their workflow better for managing any and all of the following: [code health](../strategy/use-cases/code-health.md), [code reuse](../strategy/use-cases/code-reuse.md), [dev onboarding](../strategy/use-cases/dev-onboarding.md), [fixing security vulnerabilitis](../strategy/use-cases/fixing-security-vulnerabilities.md) or [incident responce](../strategy/use-cases/incident-response.md)?

Reviewer: ```suggestion
At Sourcegraph, we aim to understand software development workflows and what matters most to allow us to make decisions about how to best serve our customers. Think of a use case as a common scenario or problem for which Sourcegraph brings forth solutions, they are the value we bring to our customers by solving a particular challenge. It is what they experience or do on our platform that makes their workflow better for managing any and all of the following: [code health](../strategy/use-cases/code-health.md), [code reuse](../strategy/use-cases/code-reuse.md), [dev onboarding](../strategy/use-cases/dev-onboarding.md), [fixing security vulnerabilitis](../strategy/use-cases/fixing-security-vulnerabilities.md) or [incident responce](../strategy/use-cases/incident-response.md)?
```

> Each use case is a challenge our customers face for which they use various processes, tools and solutions with the ultimate goal to be more effective and efficient in building _their_ products. Sourcegraph provides capabilities that solve these problems in a delightful, innovative way so we can inspire positive behavior change that benefits their team work.

Reviewer: ```suggestion
Each use case is a challenge our customers face for which they use various processes, tools, and solutions with the ultimate goal to be more effective and efficient in building _their_ products. Sourcegraph provides capabilities that solve these problems in a delightful, innovative way so we can inspire positive behavior change that benefits their teamwork.
```

- Demonstrable; tell a cohesive narrative by tying together all parts of our product
- Able to be tested discretely and completely (timeboxed) by a customer to prove out the value

# Applied in Product & Engineering

> This section provides context on how use cases align to our platform, products, features and roadmap to set forth a framework teams can leverage to make use cases first class citizens in our platform.

Reviewer: ```suggestion
This section provides context on how use cases align to our platform, products, features, and roadmap to set forth a framework teams can leverage to make use cases first-class citizens in our platform.
```

> The use cases are problems that our customers encounter that our product helps solve. In Product and Engineering, we’re building for these scenarios, demonstrating a clear progression of solving our customers (and potential customers) pain points over time. When we talk about what’s coming next on the roadmap, we think about how we’re enhancing the product to better solve these use cases. They provide general alignment and focus that the team is constantly thinking about, across the organization and within our product teams and features. When we package together Sourcegraph’s features like search, code insights, code intelligence, batch changes, notebooks - we’re able to create a cohesive and compelling narrative and solution in a particular use case for the end user.

Reviewer: ```suggestion
The use cases are problems that our customers encounter that our product helps solve. In Product and Engineering, we’re building for these scenarios, demonstrating a clear progression of solving our customers' (and potential customers') pain points over time. When we talk about what’s coming next on the roadmap, we think about how we’re enhancing the product to better solve these use cases. They provide general alignment and focus that the team is constantly thinking about, across the organization and within our product teams and features. When we package together Sourcegraph’s products like search, code insights, code intelligence, batch changes, notebooks - we’re able to create a cohesive and compelling narrative and solution in a particular use case for the end-user.
```

## How do use cases fit within our platform?

Companies have different workflows for each use case, for example how they monitor and fix security vulnerabilities. In turn, our platform provides products and features that collectively create solutions for each use case. Our customers leverage Sourcegraph in innovative ways to solve for use cases they care about, that's how we deliver value. This is how it fits together:

- Platform : the overall Sourcegraph offering
- Product : one of the vertical products in our offering (e.g. search, code monitor, code insights, batch changes)

> - Feature : a capability within a products

Reviewer: ```suggestion
- Feature : a capability within a product
```

- Solution : any collection of products and features that solve a problem or use case

> If a customer cares about Code Health, they may use Code Monitoring differently than one that cares about monitoring for Security Vulnerabilities. Our focus is to enrich our offering in each one of these products to support solving for each use case. This will make it easiery for our users to recognize where Sourcegraph could provide value in their workflows.

Reviewer: ```suggestion
If a customer cares about Code Health, they may use Code Monitoring differently than one that cares about monitoring for Security Vulnerabilities. Our focus is to enrich our offering in each one of these products to support solving for each use case. This will make it easier for our users to recognize where Sourcegraph could provide value in their workflows.
```

## How will we measure our success in the context of a use case?

We aim to increase the relevancy of our product for technology organizations that care about these challenges and want to solve them at their company. Our business measure of success will be identified through Revenue growth while on the product side this will exhibit through better engagement in our products; especially around the features that provide solutions for these use cases. By taking a more focused approach we will be able to move fast and prove the efficacy of these use cases to our growth.

> ## Are these all use cases?

Reviewer: ```suggestion
## Are these the only use cases?
```
